contracts/ContributionAccountingToken.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+// Compatible with OpenZeppelin Contracts ^5.0.0
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Permit.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+contract ContributionAccountingToken is ERC20, ERC20Permit, AccessControl {
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    
+    uint256 public maxSupply;
+    uint256 public thresholdSupply;
+    uint256 public maxExpansionRate;
+    bool public transferRestricted = true;
+    
```

Comment:
Let's add a `uint256 public immutable clowderFee`, to hold a percentage fee (e.g. 0.5%).